src/data/SeriesUtils.tsx

```diff
@@ -284,18 +287,35 @@ function buildSeries(
   let sum = 0;
   let remainder = 0;
   let binIndex = 0;
+  let nullCount = 0;
+  let nullRemainder = 0;
+  let isNull = false;
 
   for (let index = 0; index < values.length; index++) {
     sum += remainder;
+    nullCount += nullRemainder;
+
+    isNull = values[index] == null;
```

Comment:
Can this actually evaluate to `true`?  `values` is of type `number[]`, so it shouldn't be possible for it to contain a null value.  I think the comparison should be against a `0` (number) value instead

Reply:
@boazreicher Good point.  I am new to Typescript so I'm probably misunderstanding, but it seems like null values are present in my example with CSV content test datasource?   
![image](https://github.com/boazreicher/mosaic-plot/assets/711780/114e9a42-bd1c-462a-a315-54388d01d735)

If we use zero how would we distinguish from actual zero values?  For example if a sensor is measuring rain but it is not raining that would be "real" zero, but if the sensor was broken a null would be returned.

Reply:
Thinking about this more, it seems like the plugin is generally setup assuming zero is equivalent to null, so having a distinction between the two might be hard to accommodate?

I think we can ensure values are always numbers if we cast to number in createValuesArray?  Current code:
`values: Vector<any>`
...
`result.push(values.get(index));`

Reply:
So, I tested it and I believe that your change here should be fine.
I tested this:
```typescript
function createValuesArray(values: Array<any>): number[] {
  let result: number[] = []

  for (let index = 0; index < values.length; index++) {
    result.push(values[index])
  }

  return result
}

let val1 = 1
  let val2 = null
  let val3 = 2
  let result: number[] = createValuesArray([val1, val2, val3])

  let sum = 0
  for (let index = 0; index < result.length; index++) {
    console.log("index: " + index + " value: " + result[index])
    sum += result[index]
    console.log("sum is now: " + sum)
  }
```
And the output is:
```
index: 0 value: 1
sum is now: 1
index: 1 value: null
sum is now: 1
index: 2 value: 2
sum is now: 3
```

It seems (I honestly don't understand why this works...) that even though we're defining the array (and return type of `createValuesArray`) as `number[]`, they are treated as if they were something like `Array<number | null>`, and so your check `isNull = values[index] == null;` should be fine

Reply:
@andybaumgar  Since this is fine, if you can make the suggested (minor indentation, formatting) changes, I'll merge this PR and submit the updated plugin to Grafana Marketplace

Reply:
Interesting, from the [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#array_methods_and_empty_slots), apparently "empty slots" are a feature of the Array object itself.

Reply:
@boazreicher Made all the changes!